### ISSUE TITLE
Consider simpler support for apps that control all of their subdomains

### ISSUE THREAD

**Issue by brandonfancher** [EOSIO/manifest-spec#10]
The current manifest spec model treats subdomains as separate apps. This is a feature--domain owners may want to run multiple apps from their domain, or they may want to lease subdomains to others to control.

This presents quite the overhead for applications that utilize many subdomains and control all of their subdomains. They must serve a custom `chain-manifests.json` from the root of each subdomain...even if they want all of those subdomains to appear as the same app. Furthermore, serving static files like this from each subdomain often requires extra setup, like custom modifications to webpack, etc. This can hurt adoption of the standard.

For discussion, here's an improvement idea for scrutiny:

**Premise:**
A request (_e.g._, a transaction) is proposed from, say, `https://gazelle.eos.io` (_i.e._, `declaredDomain` is `https://gazelle.eos.io`).

**Current Behavior:**
Conforming authenticators would attempt to fetch `https://gazelle.eos.io/chain-manifests.json`. If it's not present, the authenticator would throw and error and not allow the request.

**Proposed Behavior:**
Conforming authenticators would attempt to fetch `https://gazelle.eos.io/chain-manifests.json`. If it's not present, the authenticator would attempt to fetch `https://eos.io/chain-manifests.json`. If it is present, the authenticator would verify that the manifest domains therein contain a wildcard (`*`) as the subdomain (_i.e._, `*.eos.io`). If there is a wildcard, that `chain-manifests.json` would be accepted and remaining checks would be done. If there is no wildcard, the manifest would be rejected.

This allows a domain to declare that they own and control all subdomains, but still gives them the flexibility to customize manifests for any subdomain they wish.

Does the security model hold?

**Comment by brandonfancher** [comment]
@bhazzard, @TaraTritt, @jeffreyssmith2nd, @Todd-Bowden, @tbfleming:

Would you guys mind reviewing the thoughts above to validate the approach from usability and security standpoints?

**Comment by tbfleming** [comment]
Explorers that watch `add.manifest` would also need to understand the wildcard in the `domain` field.

**Comment by brandonfancher** [comment]
True. I'm not aware of any explorers in existence yet. So now's a good time to be making these kinds of changes.

**Comment by TaraTritt** [comment]
I think this is a tradeoff in security for usability. I think allowing a wildcard in the manifests domain would open up an attack vector due to common [subdomain](https://www.hackerone.com/blog/Guide-Subdomain-Takeovers) [takeovers](https://labs.detectify.com/2014/10/21/hostile-subdomain-takeover-using-herokugithubdesk-more/), but i do think that it provides an easier onboarding experience to meeting the specification. Haven't fully thought out the repercussions or feasibility of this yet, but is it possible for the manifests `domain` field to allow an array of domains, which could include subdomains? It would still attempt to find the `chain-manifests.json` in the manner that you discussed, but validate against the `domain` (maybe update the name to `domains`) array to confirm that it contains the subdomain url where the request originally came from. This also might decrease the burden on app developers.
